### ISSUE TITLE
chips(readonly): chips can be selected/focused via keyboard

### ISSUE THREAD

**Issue by EamonKavanagh** [angular/material#10850]
**Actual Behavior**:
Setting `readonly="true"` on an `md-chip` component causes tab/arrow navigation to function incorrectly. In an example with 5 chips, the expected behavior would be pressing the right arrow would cycle through the chips in left to right order before wrapping back to the start.

**CodePen** (or steps to reproduce the issue): * https://material.angularjs.org/latest/demo/chips
In the same example with 5 chips, the actual behavior is:

1) First right arrow key press focuses on the first chip 
2) Second right arrow key press 'selects' the second chip without focusing on it (no visual indication of selection but selection can be verified using ChromeVox) 
3) Third right arrow key press 'selects' the first chip, with no visual indication as in 2) 
4) Forth right arrow key press focuses on first chip as in 1), cycling back to the start.

Pressing the left arrow keys does the above 4 steps in reverse so the middle chip is not accessible.

**Versions**: *
AngularJS: 1.6.4-local+sha.617b36117
AngularJS Material: 1.1.10

**Additional Information**:
Chrome 59.0.3071.115

**Comment by Splaktar** [comment]
Thank you for reporting this. I can reproduce this on the latest docs site (1.1.10).

**Comment by marosoft** [comment]
Actually what should be a default behaviour here? Should you be able to navigate through the chips if those are in readonly mode or not?

There are three lines of code in the chipsDirective.js file trying to block the navigation. However it is not blocked in the chipsController.js (left & right arrow key).

Removing the checks `!$mdChipsCtrl.readonly && ` in those lines fixes the problem with the navigation (allowing it for the readonly mode):
* https://github.com/angular/material/blob/master/src/components/chips/js/chipsDirective.js#L211
* https://github.com/angular/material/blob/master/src/components/chips/js/chipsDirective.js#L212
* https://github.com/angular/material/blob/master/src/components/chips/js/chipsDirective.js#L381

If this is what is expected, I can create a pull request for it.

**Comment by Splaktar** [comment]
For a11y, I thought that we might need to allow focusing and selecting of readonly chips as that might be the only way that a screen reader would be able to see what chips exist.

However, after doing some testing in Voiceover tonight, that isn't the case. The screen reader has a way to go through each chip using screen reader specific shortcuts (not arrows or tab).

So it looks like the actual fix here is to add more checks for `readonly` so that the keyboard interactions act more like the current click interactions (i.e. do nothing if readonly). We certainly don't want to keep the existing behavior which is quite confusing.

**Comment by Splaktar** [comment]
@marosoft thank you for asking before creating the PR. I've clarified things a bit above. It still looks like the fix is somewhat straightforward if you would like to investigate a little more and submit a PR. Thank you very much for helping to dig into this!

**Comment by marosoft** [comment]
@Splaktar I checked this again. I was playing with Voiceover (never used it before) and I could not make it work properly until I implemented further changes. 

I will create a PR soon. For now, please check this CodePen: https://codepen.io/marosoft/pen/RBKvoQ?editors=1010#0

**Comment by Splaktar** [comment]
@marosoft For some reason it's telling me "5 of 7" for Apple, but then "1 of 3" and "2 of 3" for Banana and Orange.

In some cases, it's not letting me get into the list with the ctrl-option-shift-down shortcut (odd). 

I tried the ctrl-option-down shortcut and that takes me into the chip and starts reading the "Press delete to remove this chip" message. We should probably not set that `aria-label` when `readonly`.

Thanks for the update on progress. I should have time to review your PR shortly after July 30th if you have one posted by then.

**Comment by Splaktar** [comment]
@marosoft I'm starting to look into this along with other chips a11y issues. Do you have any input on what you changed to make things a bit better?

**Comment by marosoft** [comment]
@Splaktar it was more about the logic itself rather than a11y. I will try to push my branch today so you will be able to see what I did so far.

**Comment by marosoft** [comment]
@Splaktar I pushed my branch. This is still WIP as I did not work on it after I posted the CodePen. Please check it [here].(https://github.com/marosoft/material/commit/a9db859abda9a6588e2467f753b6a4cd31649392)

**Comment by Splaktar** [comment]
PR https://github.com/angular/material/pull/11422 fixed the following issues
- problems with chip numbering being off (mentioned in OP and https://github.com/angular/material/issues/10850#issuecomment-406845542)
- readonly chips are focusable via the keyboard
